File: zenscript-code-model/src/main/java/raylras/zen/model/symbol/ClassSymbol.java
package raylras.zen.model.symbol;

import raylras.zen.model.type.ClassType;

import java.util.List;

public interface ClassSymbol extends Symbol, SymbolProvider {

    String getQualifiedName();

    String getSimpleName();

    List<Symbol> getDeclaredMembers();

    List<ClassSymbol> getInterfaces();

    @Override
    ClassType getType();

    @Override
    List<Symbol> getSymbols();

}

File: zenscript-code-model/src/main/kotlin/raylras/zen/model/symbol/ClassSymbol.kt
package raylras.zen.model.symbol

import raylras.zen.model.type.ClassType

interface ClassSymbol : Symbol, SymbolProvider {
    val qualifiedName: String

    val declaredMembers: Sequence<Symbol>

    val interfaces: Sequence<ClassType>

    override val type: ClassType
}
